coding/reader.hpp

```diff
@@ -19,10 +20,12 @@ class Reader
   DECLARE_EXCEPTION(SizeException, Exception);
   DECLARE_EXCEPTION(ReadException, Exception);
 
+  using TReaderPtr = unique_ptr<Reader>;
```

Comment:
I have to mention it's template parameter each time. And with alias I can avoid doing so.